astropy/coordinates/funcs.py

```diff
@@ -153,11 +154,14 @@ def get_sun(time):
     # We have to manually do aberration because we're outputting directly into
     # GCRS
     earth_p = earth_pv_helio[..., 0, :]
-    earth_v = earth_pv_helio[..., 1, :]
+    earth_v = earth_pv_bary[..., 1, :]
+
+    # convert barycentric velocity to units of c
+    earth_v = (earth_v*u.au/u.d).to(u.m/u.s)/c
 
     dsun = np.sqrt(np.sum(earth_p**2, axis=-1))
```

Comment:
If you add `keepdims=True` to the sum, you don't need the reshape below any more.